### ISSUE TITLE
show offline files when server connectivity is not available

### ISSUE THREAD

**Issue by alick** [haiwen/seadroid#436]
Hi,

Problem: Currently when using the Android app, if one tries to refresh when the network connection is not available, the file list view becomes blank. Nothing can be done till the connection is back.

Expectation: it would be better the file list view can show at least the (downloaded) offline files, allowing them to be read, and even edit.

Side note: the expected behavior is what is present in the ownCloud Android app.

**Comment by shoeper** [comment]
Currently usable workaround: Use any file explorer to access the cached files on your SD Card.

**Comment by alick** [comment]
@shoeper yeah, currently I use ES file explorer to access them. But I do hope seafile can show them natively, possibly also showing the filenames that have not yet been downloaded.

**Comment by freeplant** [comment]
@alick Are you sure you can always reproduce the problem? By design, Seafile caches the directory for offline use. As I tested with version 1.9.2:

1. Open a folder
2. Turn off the wiki and 3G
3. Refresh the folder, a "network unavailable" notification will appear
4. Go up to top folder
5. Go back to the folder, the content of the folder can still be shown.

**Comment by alick** [comment]
@freeplant Ah, your step 4&5 did the trick! I did not know I should go back to the top (or upper) folder.

In fact, my test or use case is like:

Considering a server is available on internal network only, and can be accessed by internal wifi or with VPN.

1. Open a folder
2. Turn off (internal) wifi, while leaving 3G on. (then the server is unavailable)
3. Refresh the folder. It shows "Error when loading the folder". And there is NO way to go back to upper folder.
4. Exit seafile and re-enter, it will try loading again, and fail with the same message.

I suggest to show something like "show offline view" in the error state in step 3&4.

**Comment by darkdragon-001** [comment]
@freeplant I am still having this issue in the latest version? Can you elaborate how one can access offline files in the latest version?

**Comment by zhwanng** [comment]
When using the latest version of Seafile v3.0.14, is it unable to display data that has already been loaded when there is no network connection, such as when WiFi and mobile data are turned off? Also, does the title bar not display the text indicating offline mode?

**Comment by zhwanng** [comment]
Yes, v2.x is no longer maintained, v3.0 was released in the previous year, the offline mode feature was added in v3.0.10. 

for the time being, the offline mode feature is not manually enabled, and the offline mode will only be activated when all networks are unavailable.

**Comment by darkdragon-001** [comment]
Thanks for the clarification. When was v3.0.10 released?

With the airplane mode workaround, I think I am fine for now. It would be great if a "force offline mode" button was added in the future.

**Comment by zhwanng** [comment]
- v3.0.10 : 2025-04-19, added offline mode feat
- v2.3.7   : 2024-05-30, v2.x the last release
- v3.0.0   : 2024-05-01, v3.x the first release
